### PR TITLE
metrics: Ignore 4xx results

### PR DESCRIPTION
**Pull request by abukosek**
Group all 4xx results under the same "ignored" label, as they're mostly just bots and we're not interested in those requests for metrics anyway.

### CODE REVIEW

File: api/middleware.go [hunk 1]
```diff
@@ -73,17 +73,30 @@ func MetricsMiddleware(m metrics.RequestMetrics, logger log.Logger) func(next ht
...
-			timer.ObserveDuration()
 
 			statusTxt := "failure"
 			if httpStatus >= 200 && httpStatus < 400 {
 				statusTxt = "success"
+			} else if httpStatus >= 400 && httpStatus < 500 {
+				// Group all 4xx errors into the "ignored" label, as they're
+				// almost always just bots and are not relevant to the metrics
+				// we're interested in.
+				statusTxt = "ignored"
```
Comment (mitjat) [inline api/middleware.go:98]: Let's do `failure_4xx` or something here. We do know the status, we're only ignoring the URL.

File: api/middleware.go [hunk 2]
```diff
@@ -73,17 +73,30 @@ func MetricsMiddleware(m metrics.RequestMetrics, logger log.Logger) func(next ht
...
+				timerName = "ignored"
```
Comment (mitjat) [inline api/middleware.go:99]: nit: Can you rename this to `metricName` or (perhaps better) `normalizedURL` or similar? It was already weird that we were using `timerName` to name a counter, and it's doubly weird now that we're using this variable more actively.

**Review by mitjat — APPROVED** [review]
Thank you!

**Comment by abukosek** [comment]
Thanks, fixed :)

**Pull request merged**